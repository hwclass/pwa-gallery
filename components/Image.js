export default class extends React.Component {
  render() {
    const { id, uri, title } = this.props
    return (
      <article>
        <picture key={id} className="pic">
          <source
            media="(min-width: 36em)"
            srcSet={`http://${uri}${process.env.IMAGE_LARGE_SIZE_SUFFIX}.jpg 1024w, http://${uri}${process.env.IMAGE_LARGE_SIZE_SUFFIX}.jpg 640w, http://${uri}${process.env.IMAGE_LARGE_SIZE_SUFFIX}.jpg 320w`}
            sizes="33.3vw"
          />
          <source
            srcSet={`http://${uri}${process.env.IMAGE_LARGE_SIZE_SUFFIX}.jpg 2x, http://${uri}${process.env.IMAGE_SMALL_SIZE_SUFFIX}.jpg 1x`}
          />
          <img src={`http://${uri}${process.env.IMAGE_LARGE_SIZE_SUFFIX}.jpg`} />
          <style jsx>{`
            .pic {
              background: #FFF;
            }
            img, src {
              margin: 0 auto;
              display: block;
            }
          `}</style>
        </picture>
        <span className="title">
          {title}
          <style jsx>{`
            .title {
              display: block;
              text-align: center;
              font-size: 2.5vmin;
            }
            @media screen and (max-width: 400px) {
              .title {
                display: none;
              }
            }
          `}</style>
        </span>
      </article>
    );
  }
}
