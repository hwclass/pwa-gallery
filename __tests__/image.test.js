/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import Image from '../components/Image'

describe('Image', () => {
  it('should render', () => {
    const component = shallow(<Image id='1' uri='/some-url/for-image-1.jpg' title='Image Title'/>)
    expect(component.getElements()).toMatchSnapshot();
  })
})

describe('Image', () => {
  it('should match with snapshot', () => {
    const component = renderer.create(<Image id='1' uri='/some-url/for-image-1.jpg' title='Image Title'/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})