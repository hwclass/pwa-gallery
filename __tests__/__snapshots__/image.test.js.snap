// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should match with snapshot 1`] = `
<article>
  <picture
    className="jsx-3329984789 pic"
  >
    <source
      className="jsx-3329984789"
      media="(min-width: 36em)"
      sizes="33.3vw"
      srcSet="http:///some-url/for-image-1.jpg_27.jpg 1024w, http:///some-url/for-image-1.jpg_27.jpg 640w, http:///some-url/for-image-1.jpg$_27.jpg 320w"
    />
    <source
      className="jsx-3329984789"
      srcSet="http:///some-url/for-image-1.jpg_27.jpg 2x, http:///some-url/for-image-1.jpg_2.jpg 1x"
    />
    <img
      className="jsx-3329984789"
      src="http:///some-url/for-image-1.jpg_27.jpg"
    />
  </picture>
  <span
    className="jsx-2158314212 title"
  >
    Image Title
  </span>
</article>
`;

exports[`Image should render 1`] = `
Array [
  <article>
    <picture
      className="jsx-3329984789 pic"
    >
      <source
        className="jsx-3329984789"
        media="(min-width: 36em)"
        sizes="33.3vw"
        srcSet="http:///some-url/for-image-1.jpg_27.jpg 1024w, http:///some-url/for-image-1.jpg_27.jpg 640w, http:///some-url/for-image-1.jpg$_27.jpg 320w"
      />
      <source
        className="jsx-3329984789"
        srcSet="http:///some-url/for-image-1.jpg_27.jpg 2x, http:///some-url/for-image-1.jpg_2.jpg 1x"
      />
      <img
        className="jsx-3329984789"
        src="http:///some-url/for-image-1.jpg_27.jpg"
      />
      <JSXStyle
        css=".pic.jsx-3329984789{background:#FFF;}img.jsx-3329984789,src.jsx-3329984789{margin:0 auto;display:block;}"
        styleId="3329984789"
      />
    </picture>
    <span
      className="jsx-2158314212 title"
    >
      Image Title
      <JSXStyle
        css=".title.jsx-2158314212{display:block;text-align:center;font-size:2.5vmin;}@media screen and (max-width:400px){.title.jsx-2158314212{display:none;}}"
        styleId="2158314212"
      />
    </span>
  </article>,
]
`;
